Executable tutorial final submission: Docker container running an FTP server

mans-andersson opened the pull request:
# Executable tutorial: Docker container running an FTP server

## Members

Kalle Meurman (kallem@kth.se)
GitHub: [Kalle](https://github.com/Wizkas0)

Måns Andersson (manande@kth.se)
GitHub: [Måns](https://github.com/mansand1)

## Proposal

We want to show how to create a Docker image that runs an FTP server (vsftpd).
The goal of the tutorial is to teach how to set up a Docker container, expose 
Docker container ports and teach the basics of how FTP works.

## Submission

Here is the link to the initial proposal PR https://github.com/KTH/devops-course/pull/1343

Our tutorial can be run in the browser and is available [here](https://katacoda.com/mansand/scenarios/dockerized_ftp_server)

The source of the tutorial can be found in the repository https://github.com/mansand1/katacoda-scenarios

_Easter egg hint: the end of the instruction for step 3_

## Grading criteria we strive for

|  Criteria                                          | Yes | No | Remarkable |
|-------------------------------------------- | ----|----|-------------|
|The TA can successful execute all the commands of the tutorial (mandatory) | **✓ Yes** | No | **✓ In the browser** |
|The tutorial gives enough background | **✓ Yes** | No | Comprehensive background |
|The tutorial is easy to follow  | **✓ Yes** | No | **✓ Well documented** |
|The tutorial contains [easter eggs](https://github.com/OrkoHunter/python-easter-eggs) | **✓ Yes** | No | **✓Subtle and fun** |
|The language is correct | **✓ Yes** | No | Interesting narrative  |

mans-andersson committed:
Add executable tutorial submission

Co-authored-by: Kalle <Wizkas0@users.noreply.github.com>

mans-andersson commented:
Hello @khaes-kth ! We have now made a submission for our executable tutorial. No students have signed up to give us feedback on our work and we both need feedback to pass the feedback task (if I understand the criteria correctly). It would be very nice if you some other TA could provide us feedback.

khesoem commented:
Hi @mansand1 ,

Here is my feedback:

# Step-by-Step

## Introduction

Very good. Concise and useful.

## Step 1

No problem found.

## Step 2

> This means that this image is based on the image centos (version 7)

Please explain what being "based" one another image means.

## Step 3

Cool!

## Step 4

Please explain --name and --restart arguments.

## Step 5

> we can **try to** upload to via the server shortly

Typo.

> Note: If you changed the username from katacoda to something else earlier, you need to change the command accordingly.

Why? We have not used it yet. After reading the rest, I realized you use them later. Please move the note somewhere that the reader can understand its importance.

# Suggestion

Overall, very smooth and useful. I would suggest that you mention that users can run commands on the running container (e.g., using the `exec` command). A tutorial to docker without this point is incomplete.

mans-andersson commented:
Hello @khaes-kth , thank you for great feedback! We have now updated our tutorial and updated this pull request to final submission. We fixed each of the things you suggested for the specific steps. We also added a section about the `docker exec` command with an example, as you suggested. This section can be found at the end of step 5.
